hkmc2/shared/src/test/mlscript-compile/FingerTreeList.mls

```diff
@@ -0,0 +1,253 @@
+import "./Option.mls"
+import "./Iter.mls"
+
+open Option
+open Iter
+open FingerTreeList
+
+fun normIdx(idx, len) =
+  if idx < 0 then idx + len
+  else idx
+
+// Follows the semantics of Array.prototype.slice indices:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
+fun normIdxSlice(i, len) =
+  if i < 0 and
+    i >= -len then len + i
+    else 0
+  else i
+
+// Node[T]
+class Branch2(val size, val e1, val e2)
+class Branch3(val size, val e1, val e2, val e3)
+
+fun getNodeSize(node) = if node is
+  Branch2(s, _, _) then s
+  Branch3(s, _, _, _) then s
+  _ then 1
+
+// View[T]
+object Nil
+class View(val e1, val rest) // T, FingerTreeList[T]
+
+// FingerTreeList[T]
+class LFT extends IterableBase with
```

Comment:
BTW why is this called `LFT`?

Reply:
Historic reasons :sweat_smile: I'll rename to FTL